drivers/mbed_reset_reason.cpp

```diff
@@ -0,0 +1,45 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mbed_reset_reason.h"
+
+#if DEVICE_RESET_REASON
```

Comment:
That's incorrect (for both h and cpp), the functions in the driver should be included all the time, so that the application don't need to check for the `DEVICE_RESET_REASON` macro. They should return some sort of error when the device doesn't support this API.